Providers/Pocket/PocketSponsoredStoriesProvider.swift

```diff
@@ -0,0 +1,127 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/
+
+import Shared
+
+class PocketSponsoredStoriesProvider: PocketSponsoredStoriesProviding, FeatureFlaggable, URLCaching, Loggable {
+
+    var endpoint: URL {
+        if featureFlags.isCoreFeatureEnabled(.useStagingSponsoredPocketStoriesAPI) {
+            return PocketSponsoredConstants.staging
+        } else {
+            return PocketSponsoredConstants.prod
+        }
+    }
+
+    lazy var urlSession: URLSession = makeURLSession(userAgent: UserAgent.defaultClientUserAgent, configuration: URLSessionConfiguration.default)
+    private lazy var pocketKey: String = {
+        return Bundle.main.object(forInfoDictionaryKey: PocketSponsoredConstants.pocketEnvAPIKey) as? String ?? ""
+    }()
+    lazy var urlCache: URLCache = {
+        return URLCache.shared
+    }()
+
+    private lazy var pocketId: String = {
+        if let pocketId = UserDefaults.standard.object(forKey: PocketSponsoredConstants.pocketId) as? String {
+            return pocketId
+        } else {
+            let uuid = UUID()
+            UserDefaults.standard.setValue(uuid.uuidString, forKey: PocketSponsoredConstants.pocketId)
+            return uuid.uuidString
+        }
+    }()
+
+    private var sponsoredFeedRequest: URLRequest? {
+        var request = Locale
+            .current
+            .regionCode?
+            .map { URLQueryItem(name: .country, value: $0) }
+            .map { endpoint.withQueryParams([$0]) }
+            .map {
+                URLRequest(
+                    url: $0,
+                    cachePolicy: .reloadIgnoringCacheData,
+                    timeoutInterval: 5
+                )
+            }
+
+        let body: [String: Any] = [
+            .version: 2,
+            .consumer_key: pocketKey,
+            .pocket_id: pocketId
+        ]
+
+        let bodyData = try? JSONSerialization.data(withJSONObject: body)
+        request?.httpBody = bodyData
+        request?.httpMethod = .post
+        return request
```

Comment:
```suggestion

        return request
```
Nit. Returns should be on their own separate line. Logically demarcating the end of the function.